File: Beginner/ATM Machine (ATM2)/atm2.java
import java.util.*;
import java.lang.*;
import java.io.*;


class Codechef {
	public static void main (String[] args) throws java.lang.Exception {
		Scanner scan=new Scanner(System.in);
		if(scan.hasNext()){
		    int t = scan.nextInt();
		    for(int p=0;p<t;p++){
		        int N = scan.nextInt();
		        int K = scan.nextInt();
		        int arr[]=new int[N];
		        for(int i=0;i<N;i++)
		            arr[i]=scan.nextInt();
		        for(int i=0;i<N;i++){
		            if(arr[i]<=K){
		                K = K-arr[i];
		                System.out.print(1);
		            }
		            else
		                System.out.print(0);
		        }
		        System.out.println();
		    }
        }
        scan.close();
	}
}

File: Beginner/ATM Machine (ATM2)/atm2.py
t = int(input())


while t:
    N, K = map(int, input().split())

    arr = list(map(int, input().split()))

    for i in arr:
        if i>K:
            print(0, end='')
        else:
            print(1, end='')
            K = K-i
    print()
    t = t-1